### PR TITLE
FEAT: Added support to deploy monorepos to github pages

### LINKED ISSUE #612: [FEAT]: Deploy monorepos
**As an EmbeddedChat developer:**

**We need to**:
Deploy the monorepos `docs`, `ui-elements`, `react`, and `layout-editor`.

**So that**:
Users can test them live.

**Acceptance Criteria**:
- [x] Deploy all mentioned monorepos on GitHub Pages. Since GitHub provides only a single URL per project (`https://rocketchat.github.io/embeddedchat`), we must use routes to display different pages such as `https://rocketchat.github.io/embeddedchat` for the `react` monorepo, `https://rocketchat.github.io/embeddedchat/layout-editor` for the `layout-editor`, and so on.

**References**:
- https://rocketchat.github.io/fuselage -> https://github.com/RocketChat/fuselage: This repo follows a similar structure for multiple monorepo deployments.
- https://www.thisdot.co/blog/deploying-multiple-apps-from-a-monorepo-to-github-pages: Provides a starting point on how to deploy.
- One of my PRs, https://github.com/RocketChat/EmbeddedChat/pull/586, can help understand how to deploy a Storybook.

Deployment can be tested on your own forked repo before raising the PR.

### PR DESCRIPTION
**Pull request by devanshkansagra**
## Acceptance Criteria fulfillment

- [x] Deploy the  monorepo to the github pages

Fixes #612

## Video/Screenshots

https://github.com/user-attachments/assets/b78b7ea9-83b8-4e05-bb10-9a3b84b7bea6

### CODE REVIEW

File: .github/workflows/deploy.yml
Comment (Spiral-Memory) [inline .github/workflows/deploy.yml:6]: Change the branch to `develop`

File: .github/workflows/deploy.yml
Comment (Spiral-Memory) [inline .github/workflows/deploy.yml:29]: Add `Storybook host URL` as an environment variable. See my PR mentioned in the issue.

**Comment by Spiral-Memory** [comment]
Thanks a lot for your contribution @devanshkansagra ❤️

File: .github/workflows/deploy.yml
Comment (devanshkansagra) [reply .github/workflows/deploy.yml:6]: Done

File: .github/workflows/deploy.yml
Comment (devanshkansagra) [reply .github/workflows/deploy.yml:29]: image

Hey @Spiral-Memory, are you talking about this??

Comment (Spiral-Memory) [reply .github/workflows/deploy.yml:29]: Yes

**Comment by Spiral-Memory** [comment]
LGTM! Let's wait for the layout editor and docs to be merged. Then we can add the docs and editor to the deployment as well.

**Comment by devanshkansagra** [comment]
Sure

**Comment by smritidoneria** [comment]
Hey @devanshkansagra , I was going through your pr and  I noticed that the deploy job in this workflow contains two actions/checkout steps. It seems redundant to check out the code twice when once should be sufficient. Could you please explain why both steps are necessary?

**Comment by Spiral-Memory** [comment]
It's fantastic to see contributors reviewing each other's code, asking questions, and learning together! Keep up the great work and continue learning 🙏

**Comment by devanshkansagra** [comment]
Hey @smritidoneria, thank you for pointing out about those redundant lines. Yesterday I was researching from the fuselage repo, this-dot article, and ai, so those lines might be added during the research and I might didn't have noticed about that redundancy. I have removed that redundant line and pushing the code. 

I have seprated build workflow and deploy workflow, because incase in future some error occured in the deploy action(deploy.yml), it will be easy to find out at which workflow the error has occured.

File: .github/workflows/deploy.yml
Comment (Spiral-Memory) [inline .github/workflows/deploy.yml:75]: If you are already building it manually with yarn build:storybook, it is unnecessary to build it again. Find a script that can avoid re-building and can be used only to publish the artifacts.

Check this out, how bitovi script works, might be helpful for you: https://github.com/bitovi/github-actions-storybook-to-github-pages/blob/main/action.yaml

Comment (devanshkansagra) [reply .github/workflows/deploy.yml:75]: Okay, let me see that

Comment (Spiral-Memory) [reply .github/workflows/deploy.yml:75]:
```
outputs:
  page_url:
    description: "The URL of the page"
    value: ${{ steps.deploy.outputs.page_url }}

- name: 'upload'
      uses: actions/upload-pages-artifact@v3
      with: 
        path: ${{ inputs.path }}

    - id: deploy
      name: Deploy to GitHub Pages
      uses: actions/deploy-pages@v4
      with:
        token: ${{ github.token }}
```

Try these with required modifications

Comment (devanshkansagra) [reply .github/workflows/deploy.yml:75]: Worked!!!!

Comment (devanshkansagra) [reply .github/workflows/deploy.yml:75]: checkout https://devanshkansagra.github.io/Embedded-Chat and https://devanshkansagra.github.io/Embedded-Chat/ui-elements

Comment (devanshkansagra) [reply .github/workflows/deploy.yml:75]: and used only one job: build-and-deploy

Comment (Spiral-Memory) [reply .github/workflows/deploy.yml:75]: Awesome ! 

Please add your new commits to this PR then

Comment (devanshkansagra) [reply .github/workflows/deploy.yml:75]: okay

**Comment by Spiral-Memory** [comment]
@devanshkansagra Layout editor has been added, you can now include that and docs in deployment

**Comment by devanshkansagra** [comment]
Sure, I will add it.

**Comment by devanshkansagra** [comment]
Hey what should I add the path? 'layout-editor'?

**Comment by devanshkansagra** [comment]
Hey @Spiral-Memory, I have added the layout editor in the deploy workflow, can you please review it?

**Comment by Spiral-Memory** [comment]
LGTM ! 

Can you once share me forked deploy URL for that ?

**Comment by devanshkansagra** [comment]
> LGTM !
> 
> Can you once share me forked deploy URL for that ?

Hey @Spiral-Memory, I did not implemented in my forked repo, instead I have created another repo so that if somethings happens my fork does not break
here's that repo url: https://github.com/devanshkansagra/EC

**Comment by Spiral-Memory** [comment]
> > LGTM !
> > Can you once share me forked deploy URL for that ?
> 
> Hey @Spiral-Memory, I did not implemented in my forked repo, instead I have created another repo so that if somethings happens my fork does not break here's that repo url: https://github.com/devanshkansagra/EC

Hey, i am not asking for the repo URL, i am asking if you could share me the URL where layout-editor is hosted now ?

**Comment by devanshkansagra** [comment]
> > > LGTM !
> > > Can you once share me forked deploy URL for that ?
> > 
> > 
> > Hey @Spiral-Memory, I did not implemented in my forked repo, instead I have created another repo so that if somethings happens my fork does not break here's that repo url: https://github.com/devanshkansagra/EC
> 
> Hey, i am not asking for the repo URL, i am asking if you could share me the URL where layout-editor is hosted now ?

https://devanshkansagra.github.io/EC/layout_editor/ this one??

**Comment by Spiral-Memory** [comment]
Yep, thanks.. looking good

**Comment by Spiral-Memory** [comment]
Now, only docs deployment is remaining.

**Comment by devanshkansagra** [comment]
Okay, once that will be completed, I will add that also

**Comment by Spiral-Memory** [comment]
It's already in the docs folder now!  
Figure out a way to host the .md files in the proper format, then add it to this PR.

Some good examples are Gitbook, Docsify, Mintlify, etc.

**Comment by devanshkansagra** [comment]
Hey, maintainers and @Spiral-Memory, can you review the changes of adding a docs site? So that I can follow up if any improvements needed.

File: packages/docs/docs/Usage/theming.md
Comment (Spiral-Memory) [inline packages/docs/docs/Usage/theming.md:6]: Remove this !

File: packages/docs/docs/Usage/ec_rc_setup.md
Comment (Spiral-Memory) [inline packages/docs/docs/Usage/ec_rc_setup.md:8]: Remove this and add the following: 

EmbeddedChat provides 'Quick Control' to simplify remote configuration of its settings, including themes, via the RC app, which can be installed in your Rocket.Chat workspace. It also aids in implementing secure login by storing the resume token as an HTTP-only cookie and integrating Rocket.Chat's OAuth login capabilities.

Comment (Spiral-Memory) [reply packages/docs/docs/Usage/ec_rc_setup.md:8]: After installation, add a new heading titled ### Usage and include the following text:

Once the Quick Control RC app is installed, navigate to App Settings, select EmbeddedChat, and you can configure EmbeddedChat properties remotely and access the other mentioned features.

File: .github/workflows/deploy.yml
Comment (Spiral-Memory) [inline .github/workflows/deploy.yml:54]: Why do we need to run `yarn run build`, I think `yarn build` will also work ?

File: packages/docs/README.md
Comment (Spiral-Memory) [inline packages/docs/README.md:14]: Modify package.json and use the script `dev`, so it can be opened with `yarn dev` to have uniformity across the project

File: packages/docs/README.md
Comment (Spiral-Memory) [inline packages/docs/README.md:22]: I think `yarn build` is fine

**Comment by Spiral-Memory** [comment]
Ensure that all links reference the documentation website and do not redirect to GitHub markdown files, unless absolutely necessary.

File: packages/docs/docs/Usage/theming.md
Comment (devanshkansagra) [reply packages/docs/docs/Usage/theming.md:6]: The title themeing guide is for the the title in sidebar

File: packages/docs/docs/Usage/ec_rc_setup.md
Comment (devanshkansagra) [reply packages/docs/docs/Usage/ec_rc_setup.md:8]: done

File: .github/workflows/deploy.yml
Comment (devanshkansagra) [reply .github/workflows/deploy.yml:54]: done

**Comment by devanshkansagra** [comment]
> Ensure that all links reference the documentation website and do not redirect to GitHub markdown files, unless absolutely necessary.

You can verify it here: https://devanshkansagra.github.io/EC/docs/

File: packages/docs/README.md
Comment (devanshkansagra) [reply packages/docs/README.md:14]: done

File: packages/docs/README.md
Comment (devanshkansagra) [reply packages/docs/README.md:22]: done

File: packages/docs/docs/Usage/theming.md
Comment (Spiral-Memory) [reply packages/docs/docs/Usage/theming.md:6]: I mean remove this : An easy-to-use, full-stack component (React.js + backend behaviors) for embedding Rocket.Chat into your web app from here

**Review by Spiral-Memory — APPROVED** [review]

**Comment by Spiral-Memory** [comment]
LGTM! Amazing work

**Pull request merged**
